rest-server/test/jobDetail.js

```diff
@@ -0,0 +1,237 @@
+                      'unKnownRetriedCount': 0
+                    },
+                    'taskCreatedTimestamp': 1522559440659,
+                    'taskCompletedTimestamp': 1522560381565,
+                    'taskServiceStatus': {
+                      'serviceVersion': 0
+                    },
+                    'containerId': 'container_e11_1522466352615_0006_01_000002',
+                    'containerHost': '10.151.40.131',
+                    'containerIp': '10.151.40.131',
+                    'containerGpus': 0,
+                    'containerLogHttpAddress': 'http://10.151.40.131:8042/node/containerlogs/container_e11_1522466352615_0006_01_000002/admin/',
+                    'containerConnectionLostCount': 0,
+                    'containerIsDecommissioning': null,
+                    'containerLaunchedTimestamp': 1522559442904,
+                    'containerCompletedTimestamp': 1522560381557,
+                    'containerExitCode': 1,
+                    'containerExitDiagnostics': 'Exception from container-launch.\nContainer id: container_e11_1522466352615_0006_01_000002\nExit code: 1\nStack trace: ExitCodeException exitCode=1: \n\tat org.apache.hadoop.util.Shell.runCommand(Shell.java:545)\n\tat org.apache.hadoop.util.Shell.run(Shell.java:456)\n\tat org.apache.hadoop.util.Shell$ShellCommandExecutor.execute(Shell.java:722)\n\tat org.apache.hadoop.yarn.server.nodemanager.DefaultContainerExecutor.launchContainer(DefaultContainerExecutor.java:212)\n\tat org.apache.hadoop.yarn.server.nodemanager.containermanager.launcher.ContainerLaunch.call(ContainerLaunch.java:302)\n\tat org.apache.hadoop.yarn.server.nodemanager.containermanager.launcher.ContainerLaunch.call(ContainerLaunch.java:82)\n\tat java.util.concurrent.FutureTask.run(FutureTask.java:266)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n\nShell output: [ERROR] EXIT signal received in yarn container, exiting ...\n\n\nContainer exited with a non-zero exit code 1\n',
+                    'containerExitType': 'UNKNOWN'
+                  }
+                ],
+                'frameworkVersion': 10
+              }
+            }
+          }
+        }
+      });
+
+    global.nock(global.launcherWebserviceUri)
+      .get('/v1/Frameworks/test_job2')
+      .reply(404, {
+        'error':'JobNotFound',
+        'message':'could not find job test_job2'
+      });
+  });
+
+  //
+  // Positive cases
+  //
+  // GET /api/v1/jobs/test_job
+  it('Case 1 (Positive): should return test_job detail info', (done) => {
+    chai.request(server)
+      .get('/api/v1/jobs/test_job')
+      .end((err, res) => {
+        expect(res, 'status code').to.have.status(200);
+        expect(res, 'json response').be.json;
+        done();
+      });
+  });
+
+  //
+  // Negative cases
```

Comment:
Please also add a negative case, as in jobSshInfo.js and jobConfig.js, in which rest-server cannot connect to launcher.

Reply:
done